tests/Unit/TestHelpers.hpp

```diff
@@ -283,3 +286,51 @@ std::string get_output(const Container& c) {
   os << c;
   return os.str();
 }
+
+/*!
+ * \ingroup TestingFramework
+ * \brief Alternative to CAPTURE that prints more digits.
+ */
+#define CAPTURE_PRECISE(variable)                                    \
+  INFO(std::scientific << std::setprecision(18) << #variable << ": " \
+       << (variable));
+
+/*!
+ * \ingroup TestingFramework
+ * \brief Calculates the derivative of an Invocable at a point x, represented by
+ * an array of doubles
+ * in the domain of map with a sixth-order finite difference method.
+ *
+ * \details Intended for use with CoordinateMaps taking the domain {xi,eta,zeta}
+ * to the range {x,y,z}. This function calculates the derivative of a range
+ * index with respect to a domain index. The range index is passed
+ * explicitly to the function as range_index, while the domain index
+ * is deduced from which element of the array dx is non-zero,
+ *
+ * \requires an array dx with a single non-zero coordinate: this
+ * will be the direction along the domain in which the derivative is taken.
+ */
+template <typename Invocable, size_t VolumeDim>
+double derivative(const Invocable& map, const size_t range_index,
```

Comment:
How about `numerical_derivative` or `numerical_deriv_approx`? A bit cumbersome, but would avoid potential conflicts with other derivative-like quantities in SpECTRE.